src/notifications.c

```diff
@@ -350,28 +406,41 @@ void ClusterSlotMigrationEvent(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64
 
     case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_STARTED:
       RedisModule_Log(RSDummyContext, "notice", "Got ASM import started event.");
-      in_asm_import = true;
-      should_filter_slots = true;
       ASM_StateMachine_StartImport(slots);
       workersThreadPool_OnEventStart();
       break;
     case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_COMPLETED:
       ASM_StateMachine_CompleteImport(slots);
-      Slots_DropCachedLocalSlots(); // Local slots have changed, drop the cache
     case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_FAILED:
       RedisModule_Log(RSDummyContext, "notice", "Got ASM import %s event.", subevent == REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_FAILED ? "failed" : "completed");
-      in_asm_import = false;
-      should_filter_slots = in_asm_trim;
       // Since importing is done in a part-time job while redis is running other commands, we notify
       // the thread pool to no longer receive new jobs, and terminate the threads ONCE ALL PENDING JOBS ARE DONE.
       workersThreadPool_OnEventEnd(false);
+      if (!IsEnterprise() && subevent == REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_COMPLETED) {
+        RedisTopologyUpdater_StopAndRescheduleImmediately(ctx);
+      }
       break;
 
     // case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_MIGRATE_STARTED:
     // case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_MIGRATE_FAILED:
     case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_MIGRATE_COMPLETED:
       ASM_StateMachine_CompleteMigration(slots);
-      Slots_DropCachedLocalSlots(); // Local slots have changed, drop the cache
+      // Start 2 timers. One for the minimal delay, and one for the maximal delay.
+      RedisModule_Log(ctx, "notice", "Got ASM migrate completed event.");
+      // Check if number of indices is 0. If so, we can start trimming immediately.
+      if (Indexes_Count() == 0) {
+        RedisModule_Log(ctx, "notice", "No indices found, enabling trimming immediately.");
+        ASM_StateMachine_StartTrim(slots);

```

Comment:
This should be called from `REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_TRIM_STARTED` notification, not when we are ready